### PR TITLE
#46 Added backend interfaces and simple realization

### PR DESCRIPTION
**Pull request by MaximSmolskiy**

### CODE REVIEW

File: app/src/main/java/com/mithridat/nonoconverter/backend/ImageConverter.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+import android.graphics.Bitmap;
+
+/**
+ * Image converting class
+ */
+class ImageConverter {
+
+    /**
+     * Method for getting field from image
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/backend/ImageConverter.java]: Don't understand

File: app/src/main/java/com/mithridat/nonoconverter/backend/Nonogram.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+ */
+class Nonogram {
+
+    /**
+     * Left and top nonogram grids
+     */
+    private ArrayList<Integer>[] _left, _top;
+
+    /**
+     * Constructor by the field
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/backend/Nonogram.java]: Don't understand

File: app/src/main/java/com/mithridat/nonoconverter/backend/NonogramSolver.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+     */
+    private boolean check() {
+
+    }
+
+    /**
+     * Implementation of simple boxes algorithm
+     *
+     * @param ind - index of row or column
+     * @param type - row or column
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/backend/NonogramSolver.java]: Is it possible to fixate somewhere, what values ​​have a column type and a row type? `enum` or `static private int`, for example?

File: app/src/main/java/com/mithridat/nonoconverter/backend/ImageConverter.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+     *
+     * @param bmp - image
+     * @param rows - number of field rows
+     * @param cols - number of field columns
+     * @return field
+     */
+    static public Field getField(Bitmap bmp, int rows, int cols) {
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/backend/ImageConverter.java]: `static public` or `public static`?

File: app/src/main/java/com/mithridat/nonoconverter/backend/ImageConverter.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+     * @return thumbnail
+     */
+    static private Field getThumbnail(Bitmap bmp, int rows, int cols, int p) {
+
+    }
+
+    /**
+     * Method for checking nonogram solvability
+     *
+     * @param thu - thumbnail
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/backend/ImageConverter.java]: Why `thu`?

File: app/src/main/java/com/mithridat/nonoconverter/backend/Field.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    private int _rows, _cols;
+
+    /**
+     * Constructor by the field sizes
+     * 
+     * @param rows - number of field rows
+     * @param cols - number of field columns
+     */
+    Field(int rows, int cols) {
+        _field = new int[rows][cols];
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/backend/Field.java:28]: Is it possible to initialize this with white color?

**Comment by maik-nack** [comment]
Name of the pull request `Added backend interfaces and simple realization`, but I don't see `simple realization`.

**Comment by maik-nack** [comment]
Is it possible to create a class with static variables for colors for Field class? You can offer your ideas.

**Review by maik-nack — APPROVED** [review]
PERFECT!!!

**Review by PawelMelnik — APPROVED** [review]
Looks cool

**Pull request merged**